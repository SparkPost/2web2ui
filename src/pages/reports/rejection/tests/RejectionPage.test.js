import React from 'react';
import { RejectionPage } from '../RejectionPage';
import { shallow } from 'enzyme';

jest.mock('src/helpers/reports');

describe('RejectionPage: ', () => {
  let props;
  let wrapper;

  beforeEach(() => {
    props = {
      loading: false,
      reportOptions: {},
      list: [
        {
          count_rejected: 65,
          rejection_category_name: 'Policy Rejection',
          rejection_category_id: 1,
          reason: '551 - Cannot Relay 105',
          domain: 'gmail.com'
        },
        {
          count_rejected: 62,
          rejection_category_name: 'Policy Rejection',
          rejection_category_id: 1,
          reason: '550 - Invalid recipient ...@... (#5.1.1)',
          domain: 'gmail.com'
        }
      ],
      aggregates: {
        count_rejected: 14,
        count_targeted: 100
      },
      refreshRejectionReport: jest.fn()
    };

    wrapper = shallow(<RejectionPage {...props} />);
  });

  it('should render', () => {
    expect(props.refreshRejectionReport).not.toHaveBeenCalled();
    expect(wrapper).toMatchSnapshot();
  });

  it('should refresh when report options reference changes', () => {
    const newReportOptions = {};
    expect(props.refreshRejectionReport).not.toHaveBeenCalled();
    wrapper.setProps({ reportOptions: newReportOptions });
    expect(props.refreshRejectionReport).toHaveBeenCalledWith(newReportOptions);
  });

  it('renders correctly when loading', () => {
    wrapper.setProps({ loading: true });
    expect(wrapper).toMatchSnapshot();
  });

  it('renders loading pannel when aggregates are still loading', () => {
    wrapper.setProps({ aggregatesLoading: true });
    expect(wrapper).toMatchSnapshot();
  });

  it('should not display top level metrics when there are no aggregates', () => {
    wrapper.setProps({ aggregates: {}});
    expect(wrapper.find('MetricsSummary')).toHaveLength(0);
  });
});
