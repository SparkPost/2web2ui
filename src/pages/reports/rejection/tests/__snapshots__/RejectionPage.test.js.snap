// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RejectionPage:  renders correctly when loading 1`] = `
<Page
  title="Rejections Report"
>
  <withRouter(Connect(ReportOptions))
    reportLoading={true}
  />
  <Connect(MetricsSummary)
    rateTitle="Rejected Rate"
    rateValue={14.000000000000002}
  >
    <strong>
      14
    </strong>
     of your messages were rejected of
     
    <strong>
      100
    </strong>
     messages targeted
  </Connect(MetricsSummary)>
  <hr />
  <PanelLoading
    accent={false}
    minHeight="400px"
  />
</Page>
`;

exports[`RejectionPage:  renders loading pannel when aggregates are still loading 1`] = `
<Page
  title="Rejections Report"
>
  <withRouter(Connect(ReportOptions))
    reportLoading={false}
  />
  <PanelLoading
    accent={false}
    minHeight="115px"
  />
  <hr />
  <DataTable
    list={
      Array [
        Object {
          "count_rejected": 65,
          "domain": "gmail.com",
          "reason": "551 - Cannot Relay 105",
          "rejection_category_id": 1,
          "rejection_category_name": "Policy Rejection",
        },
        Object {
          "count_rejected": 62,
          "domain": "gmail.com",
          "reason": "550 - Invalid recipient ...@... (#5.1.1)",
          "rejection_category_id": 1,
          "rejection_category_name": "Policy Rejection",
        },
      ]
    }
  />
</Page>
`;

exports[`RejectionPage:  should render 1`] = `
<Page
  title="Rejections Report"
>
  <withRouter(Connect(ReportOptions))
    reportLoading={false}
  />
  <Connect(MetricsSummary)
    rateTitle="Rejected Rate"
    rateValue={14.000000000000002}
  >
    <strong>
      14
    </strong>
     of your messages were rejected of
     
    <strong>
      100
    </strong>
     messages targeted
  </Connect(MetricsSummary)>
  <hr />
  <DataTable
    list={
      Array [
        Object {
          "count_rejected": 65,
          "domain": "gmail.com",
          "reason": "551 - Cannot Relay 105",
          "rejection_category_id": 1,
          "rejection_category_name": "Policy Rejection",
        },
        Object {
          "count_rejected": 62,
          "domain": "gmail.com",
          "reason": "550 - Invalid recipient ...@... (#5.1.1)",
          "rejection_category_id": 1,
          "rejection_category_name": "Policy Rejection",
        },
      ]
    }
  />
</Page>
`;
